src/Microsoft.Diagnostics.Runtime/src/Linux/ElfCoreFile.cs

```diff
@@ -73,6 +81,39 @@ private IEnumerable<ElfNote> GetNotes(ElfNoteType type)
             return ElfFile.Notes.Where(n => n.Type == type);
         }
 
+        private void LoadAuxvTable()
+        {
+            if (_auxvEntries != null)
+                return;
+
+            _auxvEntries = new Dictionary<ulong, ulong>();
+            ElfNote auxvNote = GetNotes(ElfNoteType.Aux).Single();
```

Comment:
Is there any chance the dump could have no aux notes?  (Especially if the coredump wasn't generated by create dump.  Maybe a smaller dump with less options set?)

I usually default to trying to double check these issues and not throw InvalidOperationException that Single will.  For example, maybe use FirstOrDefault and handle the null return case.  (Not sure what the right thing to do is if you can't find an aux table, but if an exception needs to be thrown I would suggest `ClrDiagnosticsException`...or whatever I named the thing that clrmd throws.  That or a bad image related exception.

Reply:
Done.